hack/check-version.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+# Get the current git branch
+branch=$(git rev-parse --abbrev-ref HEAD)
+
+# Check if the branch is "main"
+if [[ "$branch" == "main" ]]; then
+    version="latest"
+# Otherwise, use the branch name as the version
+else
+    version="$branch"
+fi
+
+# Use yq to set build.version to version variable
+yq eval ".build.version = \"$version\"" -i build.yaml
```

Comment:
I see this modifies the `build.yaml` file, setting the version to the name of the branch when it's not `main`. This is indeed what I had in mind. What I missed though is that this will leave a dirty change we do not want to commit.

Not sure the best to solve this. We want the branch name version in the binary, but not in the branch itself.

Reply:
Actually, this is accounted for within `hack/check-git-dirty.sh` . https://github.com/Kuadrant/authorino/blob/b2311c1825c1006cf977648ef72949144b6998b2/hack/check-git-dirty.sh#L11

Reply:
Indeed it is but only for the purpose of setting the "dirty" flag. The dirty change to the file will remain on disk. It's a tracked file. People adding it to stage, committing the change, others overlooking it in the PRs and the thing ending up in main is only a matter of time. Believe me.

Reply:
I can see what you mean... if this is something you would rather not see, should we revert back to using a variable instead? 

I know before you said you didn't want to as it would have implications upstream, but to avoid this from becoming an issue it could be the better option? 

Let me know your thoughts...